### PR TITLE
Create a new helm chart for Vault deployment

### PR DESCRIPTION
**Pull request by revit13**
Create a new vault helm chart as described in #632

### CODE REVIEW

File: charts/Makefile [hunk 1]
```diff
@@ -25,19 +25,14 @@ vault:
...
 	kubectl create namespace m4d-system || true
 	helm repo add hashicorp https://helm.releases.hashicorp.com
-	helm install vault hashicorp/vault --version 0.9.1 \
-                --set "global.openshift=$(WITH_OPENSHIFT)" \
-                --set injector.enabled=false \
-                --set "server.dev.enabled=true" \
-		--values ../third_party/vault/plugin-secrets-kubernetes-reader/values.yaml \
+	helm dependency update vault
+	helm install vault vault \
+                --set "vault.global.openshift=$(WITH_OPENSHIFT)" \
```
Comment (roee88) [inline charts/Makefile:30]: I think that global is special and is shared with all subcharts so it should be set at top level, are we sure this one works?

File: charts/vault/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+name: vault
+description: Vault Helm Chart
+type: application
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.0.0
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application.
+appVersion: latest
```
Comment (roee88) [inline charts/vault/Chart.yaml:14]: Would it make sense not to update it to the M4D version but rather match it to the appVersion of the subchart?

**Review by roee88 — COMMENTED** [review]
Can you add README to describe the different values files?

File: charts/vault/Chart.yaml (hunk 2 above)
Comment (revit13) [reply charts/vault/Chart.yaml:14]: yes, it makes sense. I will change that `appVersion: 1.6.2` which is the vault appVersion for Vault Version 0.9.1, leaving `version: 0.0.0.`

File: charts/Makefile (hunk 1 above)
Comment (revit13) [reply charts/Makefile:30]: I checked that and it worked ok,.. for the testing I had to enable the injector to validate that the resources are generated as expected as openshift flag affects the injector deployment.

File: Makefile [hunk 3]
```diff
@@ -33,9 +33,9 @@ run-integration-tests:
 	$(MAKE) docker
 	$(MAKE) -C test/services docker-build docker-push
 	$(MAKE) cluster-prepare-wait
-	$(MAKE) configure-vault
 	$(MAKE) -C charts m4d
 	$(MAKE) -C manager wait_for_manager
+	$(MAKE) configure-vault
```
Comment (simanadler) [inline Makefile:38]: Changes here need to be reflected in the web site contributing page as well.

Comment (revit13) [reply Makefile:38]: Thanks will update the web page as well then.

**Review by roee88 — APPROVED** [review]

**Pull request merged**